src/nti/schema/field.py

```diff
@@ -361,7 +359,14 @@ def __call__(self, value):
         if self.fromUnicode is not None:
             # Hoping for strings at this point.
             # for BWC, we always call this method.
-            return self.fromUnicode(value)
+            try:
+                return self.fromUnicode(value)
+            except AttributeError:
```

Comment:
This is the area I'm not quite sure about. If we have `None` here, we'll still fall through and raise a `WrongContainedType` error.  Previously, nti.externalization would only go down this path if it had an actual value (non None) to convert.

I think this PR is correct, but I think (?) we'll want to make sure nti.externalization does not go down this code path for a `None` value.

https://github.com/NextThought/nti.externalization/blob/master/src/nti/externalization/internalization/fields.py#L289

Reply:
To be more explicit, previously we had `None` in a non-required field. nti.externalization would not try to use `fromObject`, and validation would pass because the field was not required. We will want to be able to differentiate from no value given and a field is not required (continue on, all is good) and the case were we have a invalid value and conversion fails (the user tried to give us a value but gave us invalid data).

Reply:
I propose a second PR to update `Variant`'s (and `_SequenceFromObjectMixin's) handling of `None` and `self.missing_value` in `fromObject`

Reply:
That sounds good; would they essentially just ignore `None` values?

Reply:
I think their `fromObject` methods probably start out similarly to the `validate` method of a Field:
```python
        if value == self.missing_value: # self.missing_value defaults to None
            if self.required:
                 raise RequiredMissing()
```

From there the logic differs a bit. I suspect Variant wants to still run in through all the fields to see if somebody wants to allow it (e.g., `nti.schema.Int` explicitly allows None), but it only raises the VariantValiadtionError at the end if `value != self.missing_value`.

Sequences are probably currently raising a TypeError because they can't iterate, that probably needs to be cleaned up and ignored if `value == self.missing_value`.